pkg/controllers/node/drift.go

```diff
@@ -36,20 +38,39 @@ type Drift struct {
 }
 
 func (d *Drift) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisioner, node *v1.Node) (reconcile.Result, error) {
-	if _, ok := node.Annotations[v1alpha5.VoluntaryDisruptionAnnotationKey]; ok {
+	// If the node is marked as voluntarily disrupted by another controller, do nothing.
+	val, hasAnnotation := node.Annotations[v1alpha5.VoluntaryDisruptionAnnotationKey]
```

Comment:
```suggestion
	val, ok := node.Annotations[v1alpha5.VoluntaryDisruptionAnnotationKey]
```

nit: I think typical best-practice is to use `ok` vs `hasAnnotation` for existence.